src/hooks/useSearchCurrencyList.ts

```diff
@@ -1,473 +1,243 @@
-  } else {
-    return (
-      filterList(searchList, formattedQuery, keys, {
-        threshold: isAddress ? rankings.CASE_SENSITIVE_EQUAL : rankings.CONTAINS,
-      }) || []
-    );
-  }
+type SearchItemWithRelevance = SearchAsset & {
+  relevance: number;
 };
 
-const useSearchCurrencyList = (searchQuery: string, searchChainId = ChainId.mainnet) => {
-  const previousChainId = usePrevious(searchChainId);
+const sortForDefaultList = (tokens: SearchAsset[]) => {
+  const curated = tokens.filter(asset => asset.highLiquidity && asset.isRainbowCurated && asset.icon_url);
+  return curated.sort((a, b) => (b.market?.market_cap?.value || 0) - (a.market?.market_cap?.value || 0));
+};
 
-  const searching = useMemo(() => searchQuery !== '' || ChainId.mainnet !== searchChainId, [searchChainId, searchQuery]);
+const sortTokensByRelevance = (tokens: SearchAsset[], query: string): SearchItemWithRelevance[] => {
+  const normalizedQuery = query.toLowerCase().trim();
+  const tokenWithRelevance: SearchItemWithRelevance[] = tokens.map(token => {
+    const normalizedTokenName = token.name.toLowerCase();
+
+    const normalizedTokenSymbol = token.symbol.toLowerCase();
+    const tokenNameWords = normalizedTokenName.split(' ');
+    const relevance = getTokenRelevance({
+      token,
+      normalizedTokenName,
+      normalizedQuery,
+      normalizedTokenSymbol,
+      tokenNameWords,
+    });
+    return { ...token, relevance };
+  });
 
-  const { favorites: favoriteAddresses, favoritesMetadata: favoriteMap } = useFavorites();
-  const curatedMap = rainbowTokenList.CURATED_TOKENS;
-  const unfilteredFavorites = Object.values(favoriteMap).filter(token => token.networks[searchChainId]);
+  return tokenWithRelevance.sort((a, b) => b.relevance - a.relevance);
+};
 
-  const [loading, setLoading] = useState(true);
-  const [favoriteAssets, setFavoriteAssets] = useState<RainbowToken[]>([]);
-  const [importedAssets, setImportedAssets] = useState<RainbowToken[]>([]);
-  const [highLiquidityAssets, setHighLiquidityAssets] = useState<RainbowToken[]>([]);
-  const [lowLiquidityAssets, setLowLiquidityAssets] = useState<RainbowToken[]>([]);
-  const [verifiedAssets, setVerifiedAssets] = useState<RainbowToken[]>([]);
-  const [fetchingCrosschainAssets, setFetchingCrosschainAssets] = useState(false);
-  const [crosschainVerifiedAssets, setCrosschainVerifiedAssets] = useState<CrosschainVerifiedAssets>({
-    [ChainId.apechain]: [],
-    [ChainId.arbitrum]: [],
-    [ChainId.avalanche]: [],
-    [ChainId.base]: [],
-    [ChainId.bsc]: [],
-    [ChainId.blast]: [],
-    [ChainId.degen]: [],
-    [ChainId.gnosis]: [],
-    [ChainId.gravity]: [],
-    [ChainId.ink]: [],
-    [ChainId.linea]: [],
-    [ChainId.mainnet]: [],
-    [ChainId.optimism]: [],
-    [ChainId.polygon]: [],
-    [ChainId.sanko]: [],
-    [ChainId.scroll]: [],
-    [ChainId.zksync]: [],
-    [ChainId.zora]: [],
-  });
+// higher number indicates higher relevance
+const getTokenRelevance = ({
```

Comment:
new relevance score similar to cmd K relevance score on BX; basically prioritizing name/symbol exact matches